access_error.go

```diff
@@ -38,12 +38,16 @@ func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 
 	rfcerr := ErrorToRFC6749Error(err)
 	if !f.SendDebugMessagesToClients {
-		rfcerr.Debug = ""
+		rfcerr = rfcerr.Sanitize()
 	}
 
-	js, err := json.Marshal(rfcerr)
+	js, err := json.MarshalIndent(rfcerr, "", "\t")
 	if err != nil {
-		http.Error(rw, fmt.Sprintf(`{"error": "%s"}`, err.Error()), http.StatusInternalServerError)
+		if f.SendDebugMessagesToClients {
+			http.Error(rw, fmt.Sprintf(`{\n\t"error": "server_error",\n\t"error_description": "%s"\n}`, err.Error()), http.StatusInternalServerError)
```

Comment:
If `err.Error()` contains a quote this will result in invalid JSON because of missing escaping. You might want to use tidwall/gjson or tidwall/sjson to properly escape the string.

Reply:
So this is tricky, so we get here ONLY if json encoding failed. And so I am not sure what is the point of trying another library at this point. There is something strange happening, so I attempt to do something light.

Maybe I could just do find/replace on `"` into `\"` and `\` into `\\` just to handle the most obvious cases and this is it?